### PR TITLE
update cicd docs

### PR DESCRIPTION
**Pull request by kolodziejczak**
**Description**

Changes proposed in this pull request:

- update cicd docs

### CODE REVIEW

File: docs/contributor/04-30-ci-cd.md [hunk 1]
```diff
@@ -13,38 +13,38 @@ There are two environments configured:
...
+| [`PR Integration / Integration tests with Istio module`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L55) | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs the Istio module using the release manifest).                 |
+| [`pull-api-gateway-manager-integration-gardener TBD`]()                                                                                                     | Executes the main integration test suite for the API Gateway module on a Gardener cluster.                                                                   |
+| [`PR Integration / Upgrade tests (pull_request) TBD`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L74)    | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version. |
```
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:24]: It'S already there, I think we can remove the TBD
```suggestion
| [`PR Integration / Upgrade tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L74)    | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version. |
```

File: docs/contributor/04-30-ci-cd.md [hunk 2]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
 - 'internal' - used when a repository collaborator runs a job in the repository
 - 'restricted' - used when an outside collaborator runs a job in the repository. The run must be approved by @kyma-project/goat.
 
-## Pipelines running on pull requests
+## Pipelines running on pull requests to main
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:12]:
```suggestion
## Pipelines running on pull requests to the `main` branch
```

File: docs/contributor/04-30-ci-cd.md [hunk 3]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Wait for image build`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L24)                | Waits for the operator image build to finish so that the image-dependent jobs can be executed.                                                               |
+| [`pull-api-gateway-manager-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L6)            | Builds Kyma API Gateway Operator image and pushes it to the `dev` registry.                                                                                  |
+| [`pull-api-gateway-module-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L55)            | Builds ModuleTemplate for the API Gateway module and pushes it to the `dev` registry.                                                                        |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:20]:
```suggestion
| [`pull-api-gateway-module-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L56)            | Builds ModuleTemplate for the API Gateway module and pushes it to the `dev` registry.                                                                        |
```

File: docs/contributor/04-30-ci-cd.md [hunk 4]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Integration tests`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L36)                   | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs the Istio module using Kyma CLI).                             |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:21]:
```suggestion
| [`PR Integration / Integration tests`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L36)                   | Executes the main integration test suite for the API Gateway module on a k3d cluster The job installs the Istio module using Kyma CLI.                             |
```

File: docs/contributor/04-30-ci-cd.md [hunk 5]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Integration tests with Istio module`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L55) | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs the Istio module using the release manifest).                 |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:22]:
```suggestion
| [`PR Integration / Integration tests with Istio module`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L55) | Executes the main integration test suite for the API Gateway module on a k3d cluster. The job installs the Istio module using the release manifest.                 |
```

File: docs/contributor/04-30-ci-cd.md [hunk 6]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`pull-api-gateway-manager-integration-gardener TBD`]()                                                                                                     | Executes the main integration test suite for the API Gateway module on a Gardener cluster.                                                                   |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:23]:
```suggestion
| `pull-api-gateway-manager-integration-gardener TBD`                                                                                                     | Executes the main integration test suite for the API Gateway module on a Gardener cluster.                                                                   |
```
If we're not adding the link yet, let's remove the placeholder

File: docs/contributor/04-30-ci-cd.md [hunk 7]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Upgrade tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L74)        | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version. |
+| [`Pull Request / Golang lint check (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L24)          | Performs linting.                                                                                                                                            |
+| [`Pull Request / Run unit tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L40)             | Runs unit tests with code coverage information.                                                                                                              |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:26]:
```suggestion
| [`Pull Request / Run unit tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L41)             | Runs unit tests with code coverage information.                                                                                                              |
```

File: docs/contributor/04-30-ci-cd.md [hunk 8]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Wait for image build`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L24)                                         | Waits for the operator image build to finish so that the image-dependent jobs can be executed.                                                                             |
+| [`pull-api-gateway-manager-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L6)                                     | Builds Kyma API Gateway Operator image and pushes it to the `dev` registry.                                                                                                |
+| [`pull-api-gateway-module-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L55)                                     | Builds ModuleTemplate for the API Gateway module and pushes it to the `dev` registry.                                                                                      |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:37]:
```suggestion
| [`pull-api-gateway-module-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L56)                                     | Builds ModuleTemplate for the API Gateway module and pushes it to the `dev` registry.                                                                                      |
```

File: docs/contributor/04-30-ci-cd.md [hunk 9]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Integration tests`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L36)                                            | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs the Istio module using Kyma CLI).                                           |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:38]:
```suggestion
| [`PR Integration / Integration tests`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L36)                                            | Executes the main integration test suite for the API Gateway module on a k3d cluster. The job installs the Istio module using Kyma CLI.                                           |
```

File: docs/contributor/04-30-ci-cd.md [hunk 10]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Integration tests with Istio module`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L55)                          | Executes the main integration test suite for the API Gateway module on a k3d cluster (installs the Istio module using the release manifest).                               |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:39]:
```suggestion
| [`PR Integration / Integration tests with Istio module`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L55)                          | Executes the main integration test suite for the API Gateway module on a k3d cluster. The job installs the Istio module using the release manifest.                               |
```

File: docs/contributor/04-30-ci-cd.md [hunk 11]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+| [`PR Integration / Upgrade tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-integration.yaml#L74)                                 | Executes the upgrade integration test suite for the API Gateway module and verifies if the existing release can be successfully upgraded to the new version.               |
+| [`Pull Request / Golang lint check (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L24)                                   | Performs linting.                                                                                                                                                          |
+| [`Pull Request / Run unit tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L40)                                      | Runs unit tests with code coverage information.                                                                                                                            |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:42]:
```suggestion
| [`Pull Request / Run unit tests (pull_request)`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/pull-request.yaml#L41)                                      | Runs unit tests with code coverage information.                                                                                                                            |
```

File: docs/contributor/04-30-ci-cd.md [hunk 12]
```diff
@@ -9,42 +9,61 @@ There are two environments configured:
...
+|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [`Main Integration / Wait for image build`](https://github.com/kyma-project/api-gateway/blob/main/.github/workflows/main-integration.yaml#L18)                                        | Waits for the operator image build to finish so that the image-dependent jobs can be executed.                                                                             |
+| [`post-api-gateway-manager-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L194)                                    | Builds Kyma API Gateway Operator image and pushes it to the `prod` registry.                                                                                               |
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:55]:
```suggestion
| [`post-api-gateway-manager-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/api-gateway/api-gateway-manager-build.yaml#L196)                                    | Builds Kyma API Gateway Operator image and pushes it to the `prod` registry.                                                                                               |
```

**Pull request merged**